tests/functional/commands/test_db_dump.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+"""Perform functional tests for db dump."""
+
+import os
+import pickle
+
+import orion.core.cli
+from orion.core.io.database.pickleddb import PickledDB
+from orion.storage.base import setup_storage
+
+
+def execute(command, assert_code=0):
+    """Execute orion command and return returncode"""
+    returncode = orion.core.cli.main(command.split(" "))
+    assert returncode == assert_code
+
+
+def clean_dump(dump_path):
+    """Delete dumped files."""
+    for path in (dump_path, f"{dump_path}.lock"):
+        if os.path.isfile(path):
+            os.unlink(path)
+
+
+def test_dump_default(three_experiments_family_same_name_with_trials, capsys):
+    """Test dump with default arguments"""
+    assert not os.path.exists("dump.pkl")
+    try:
+        execute("db dump")
+        assert os.path.isfile("dump.pkl")
+        dumped_db = PickledDB("dump.pkl")
+        with dumped_db.locked_database(write=False) as internal_db:
+            collections = set(internal_db._db.keys())
+        assert collections == {"experiments", "algo", "trials", "benchmarks"}
+        assert len(dumped_db.read("experiments")) == 3
+        assert len(dumped_db.read("algo")) == 3
+        assert len(dumped_db.read("trials")) == 24
+        assert len(dumped_db.read("benchmarks")) == 3
```

Comment:
I think this test is too superficial. It should verify that the links between collections are not broken. That is, algo and trials referencing experiment ids, and experiment referencing parent ids (in field `refers`).

Reply:
It should also look for trial.parent. Right now it should not matter because the trial hash does not depend on the experiment id. But if we ever make changes that affects this we would like this test to fail so that we fix the dump script accordingly.

Reply:
Done in `test_db_dump` ! To do in `test_db_load`.